### PR TITLE
Assignment 1 Gloria Domingo

### PR DESCRIPTION
**Pull request by gloriadomingo**
Assignment 1 Gloria Domingo

### CODE REVIEW

File: Assignment4/gloriadomingo-22C026/task06.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# Visualize the results
+g.add((ns.Researcher, RDFS.subClassOf, ns.Person))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+# Visualize the results
+g.add((ns.JaneSmith, RDFS.Literal, ns.Researcher))
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task06.py:50]: that's not the right relationship to declare individuals

File: Assignment4/gloriadomingo-22C026/task06.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+g.add((ns.JohnSmith, ns2.works_at, ns.UPM))
+for s, p, o in g:
+  print(s,p,o)
+
+"""**Task 6.6: Add that Jown knows Jane using the FOAF vocabulary**"""
+
+# TO DO
+# Visualize the results
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+g.add((ns.Jown, ns.knowsJaneuses, ns.FOAFvocabulary))
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task06.py:82]: What is `knowsJaneuses`?? it's not in the FOAF namespace. Also,  ns.FOAFvocabulary is unrelated!

File: Assignment4/gloriadomingo-22C026/task07.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+# TO DO
+# Visualize the results
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.orq/2001/vcard-rdf/3.0#")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print(s)
+q1 = prepareQuery("SELECT ?x WHERE {?x rdfs:subClassOf ns:LivingThing}", initNs = {"rdfs": RDFS, "ns": ns})
+for r in g.query(q1):
+  print(r)
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task07.py:33]: missing all subclasses of Living thing

File: Assignment4/gloriadomingo-22C026/task07.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+"""
+
+# TO DO
+# Visualize the results
+for persona, p, o in g.triples((None,RDF.type, ns.Person)):
+  print(persona)
+for subclase, p ,o in g.triples((None, RDFS.subClassOf,ns.Person)):
+  for persona, p, o in g.triples((None, RDF.type,subclase)):
+    print(persona)
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task07.py:45]: you need to make this method recursive.

File: Assignment4/gloriadomingo-22C026/task07.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+# Visualize the results
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.orq/2001/vcard-rdf/3.0#")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print(s)
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task07.py:30]: missing all subclasses. You need to make this method recursive

File: Assignment4/gloriadomingo-22C026/task07.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+# TO DO
+# Visualize the results
+for elem,p,o in g.triples((None, RDF.type, ns.Animal)):
+  for s,t,u in g.triples((elem, None,None)):
+    print(s,t,u)
+q1 = prepareQuery('''
+  SELECT ?x ?relacion
+  WHERE{
+    ?x rdf:type ns:Animal.
+    ?x ?relacion ?a
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task07.py:63]: What about Person?

File: Assignment4/gloriadomingo-22C026/task07.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+# Visualize the results
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+for persona,p,o in g.triples((None, RDF.type, ns.Person)):
+  for a,q,r in g.triples((None, FOAF.knows, persona)):
+    print(a)
+q1 = prepareQuery('''
+  SELECT DISTINCT ?x
+  WHERE{
+    ?x rdf:type ns:Person.
+    ?x FOAF:knows ns:RockySmith
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task07.py:81]: you are not listing the name of the person

File: Assignment4/gloriadomingo-22C026/task07.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+lista_def = []
+for n,s, o in g.triples((None, FOAF.knows, None)):
+  if s in lista:
+    lista_def += [s]
+  lista += [s]
+print(lista_def)
+q1 = prepareQuery('''
+  SELECT DISTINCT ?x
+  WHERE{
+    ?x FOAF:knows ?y
```
Comment (dgarijo) [inline Assignment4/gloriadomingo-22C026/task07.py:101]: this lists all the entities who know another. This is not what was requested

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address these problems

**Comment by anaigmo** [comment]
Entregado con errores, no se admiten más PR

**Pull request merged**